app/utils/store.py

```diff
@@ -31,17 +37,29 @@ def put(self, service_id, document_stream, *, mimetype, verification_email=None)
 
         extra_kwargs = {}
         if verification_email:
+            # Make super sure that the email is in our standardised format
+            try:
+                verification_email = clean_and_validate_email_address(verification_email)
```

Comment:
Are we callng `clean_and_validate_retention_period()` twice? Here and in  _get_upload_document_request_data()? Wouldn't it be enough to call it just here?

Reply:
Yep, definitely _should_ be enough to do it in one place! I don't mind the redundancy be because it allows this function to perhaps be called from somewhere else. It's redundant but I don't think it's problematic? :)

Reply:
Also - it's arguable that these are two different semantic calls that just happen to have the same underlying code.

1) The view needs to validate that the data it's received is valid and return clean, useful error messages if not.
2) The store is the point that controls how the email address is translated into a hash and how the hash is validated, so it needs to ensure the incoming value is in a standardised format.

Both of these use the same underlying formatting/cleaning function, but they fill different roles and _could_ theoretically be implemented differently.

Reply:
You are right that they are two separate functions, but at the same time we only call `document_store.put()` in one place that already verifies and cleans the email. I'd argue that if potentially in the future someone calls `document_store.put()` from somewhere else it is on them to make sure the input is validated.

I can't think of other places where we deliberately validate the same thing twice in the same app and flow in case someone builds something in the future and does a bad job of it.

Reply:
Alright, I'll remove it from the document store 🤷